windows/code/DBStorage.cpp

```diff
@@ -187,160 +177,158 @@ namespace
         return result;
     }
 
-    // Checks if sqliteResult is SQLITE_OK. If not, reports the error via
-    // callback & returns false.
-    bool CheckSQLiteResult(sqlite3 *db, const DBStorage::Callback &callback, int sqliteResult)
+    // Check if sqliteResult is SQLITE_OK.
+    // If not, report the error to the errorHandler and return std::nullopt.
+    std::optional<bool> CheckSQLiteResult(sqlite3 *db,
+                                          DBStorage::ErrorHandler &errorHandler,
+                                          int sqliteResult) noexcept
     {
         if (sqliteResult == SQLITE_OK) {
             return true;
         } else {
-            InvokeError(callback, sqlite3_errmsg(db));
-            return false;
+            return errorHandler.AddError(sqlite3_errmsg(db));
         }
     }
 
-    using Statement = std::unique_ptr<sqlite3_stmt, decltype(&sqlite3_finalize)>;
+    using StatementPtr = std::unique_ptr<sqlite3_stmt, decltype(&sqlite3_finalize)>;
 
-    // Creates a prepared SQLite statement. On error, returns nullptr
-    Statement PrepareStatement(sqlite3 *db, DBStorage::Callback &callback, const char *stmt)
+    // A convenience wrapper for sqlite3_prepare_v2 function.
+    int PrepareStatement(sqlite3 *db,
+                         const std::string &statementText,
+                         sqlite3_stmt **statement) noexcept
     {
-        sqlite3_stmt *pStmt{nullptr};
-        if (!CheckSQLiteResult(db, callback, sqlite3_prepare_v2(db, stmt, -1, &pStmt, nullptr))) {
-            return {nullptr, sqlite3_finalize};
-        }
-        return {pStmt, &sqlite3_finalize};
+        return sqlite3_prepare_v2(db, statementText.c_str(), -1, statement, nullptr);
     }
 
-    // Binds the index-th variable in this prepared statement to str.
-    bool BindString(sqlite3 *db,
-                    const DBStorage::Callback &callback,
-                    const Statement &stmt,
-                    int index,
-                    const std::string &str)
+    // A convenience wrapper for sqlite3_bind_text function.
+    int BindString(StatementPtr &statement, int index, const std::string &str) noexcept
     {
-        return CheckSQLiteResult(
-            db, callback, sqlite3_bind_text(stmt.get(), index, str.c_str(), -1, SQLITE_TRANSIENT));
+        return sqlite3_bind_text(statement.get(), index, str.c_str(), -1, SQLITE_TRANSIENT);
     }
 
-    struct slim_shared_lock_guard {
-        explicit slim_shared_lock_guard(winrt::slim_mutex &m) noexcept : m_mutex(m)
-        {
-            m_mutex.lock_shared();
-        }
-
-        ~slim_shared_lock_guard() noexcept
-        {
-            m_mutex.unlock_shared();
+    // Merge source into destination.
+    // It only merges objects - all other types are just clobbered (including arrays).
+    void MergeJsonObjects(winrt::Windows::Data::Json::JsonObject const &destination,
+                          winrt::Windows::Data::Json::JsonObject const &source) noexcept
```

Comment:
Nit: Use the aliases you created earlier?
```suggestion
    void MergeJsonObjects(winrt::JsonObject const &destination,
                          winrt::JsonObject const &source) noexcept
```